assets/src/scripts/components/hydrograph/store.js

```diff
@@ -129,20 +140,80 @@ export const timeSeriesReducer = function (state={}, action) {
                 }
             };
 
-
         case 'SET_MEDIAN_STATISTICS':
             return {
                 ...state,
                 tsData: {
                     ...state.tsData,
-                    medianStatistics: action.medianStatistics
+                    medianStatistics: action.medianStatistics.values
                 },
                 showSeries: {
                     ...state.showSeries,
                     medianStatistics: true
+                },
+                statisticalMetaData: {
+                    ...state.statisticalMetaData,
+                    beginYear: action.medianStatistics.beginYear,
+                    endYear: action.medianStatistics.endYear
+                }
+            };
+
+        case 'SET_LEGEND_MARKERS':
+            let marker;
+            let text;
+            if (action.key === 'compare' || action.key === 'current') {
+                text = 'Current Year';
+                let domId = `ts-${action.key}`;
+                let svgGroup = `${action.key}-line-marker`;
+                if (action.key === 'compare') {
+                    text = 'Last Year';
+                }
+                marker = defineLineMarker(domId, 'line', text, svgGroup);
+            }
+            else if (action.key === 'medianStatistics') {
+                let beginYear;
+                let endYear;
+                try {
+                    beginYear = state.statisticalMetaData.beginYear;
+                    endYear = state.statisticalMetaData.endYear;
+                    text = `Median Discharge ${beginYear} - ${endYear}`;
+                }
+                catch(err) {
+                    beginYear = '';
+                    endYear = '';
+                    text = 'Median Discharge';
+                }
+                marker = defineCircleMarker(4, null, 'median-data-series', text, 'median-circle-marker');
+            }
+            else {
+                marker = null;
+            }
+            return {
+                ...state,
+                legendMarkers: {
+                    ...state.legendMarkers,
+                    [action.key]: marker
                 }
             };
 
+        case 'SELECT_DISPLAY_MARKERS':
+            let displayMarkers = [];
+            const showSeries = state.showSeries ? state.showSeries : {};
+            const markers = state.legendMarkers ? state.legendMarkers : {};
+            for (const [key, value] of Object.entries(showSeries)) {
+                let marker;
+                if (value) {
+                    marker = markers[key];
+                    if (marker) {
+                        displayMarkers.push(marker);
+                    }
+                }
+            }
+            return {
+                ...state,
+                displayMarkers: displayMarkers
+            };
+
```

Comment:
I'm wondering if rather than save the markers in the state, we should have a selector that calculates them based on what's currently visible. In an idealized situation, the reducer should update the state with the minimal amount of data to represent the situation.

Reply:
You know, I just made a very similar comment at around the same time this was posted. :ghost:

Reply:
I just saw that 👍